commit 4ff4042fd37f8e91cde10063071cf1e20512ff4f (refs/changes/79/2279/1)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2017-07-25T12:22:08-07:00 (2 years, 2 months ago)
    
    IMGAPI-635 imgapi crashing on 'docker pull holzi/quine-relay:latest'

diff --git a/CHANGES.md b/CHANGES.md
index 766c04b..bf0cae5 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,9 @@
 # IMGAPI changelog
 
+## 4.0.1
+
+- IMGAPI-635: imgapi crashing on 'docker pull holzi/quine-relay:latest'
+
 ## 4.0.0
 
 - DOCKER-929 Support the docker v2.2 manifest format.
diff --git a/lib/docker.js b/lib/docker.js
index 82d9538..392e04f 100644
--- a/lib/docker.js
+++ b/lib/docker.js
@@ -757,12 +757,18 @@ function _dockerActivateImage(newImage, ctx, callback) {
  */
 function historyEntryFromImageJson(imgJson) {
     assert.object(imgJson.container_config, 'imgJson.container_config');
-    assert.arrayOfString(imgJson.container_config.Cmd,
-        'imgJson.container_config.Cmd');
+
+    // IMGAPI-635 imgJson.container_config.Cmd can be null.
+    var cmd = '';
+    if (imgJson.container_config.Cmd) {
+        assert.arrayOfString(imgJson.container_config.Cmd,
+            'imgJson.container_config.Cmd');
+        cmd = imgJson.container_config.Cmd.join(' ');
+    }
 
     var entry = {
         created: imgJson.created,
-        created_by: imgJson.container_config.Cmd.join(' ')
+        created_by: cmd
     };
 
     if (isMetadataCmd(entry.created_by)) {
diff --git a/package.json b/package.json
index c0c790e..e5c722a 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgapi",
   "description": "Image API to manage images for SDC 7",
-  "version": "4.0.0",
+  "version": "4.0.1",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
